app/common/model/src/test/java/io/syndesis/common/model/support/EquivalencerTest.java

```diff
@@ -0,0 +1,129 @@
+
+        return new Connection.Builder()
+                       .id("5")
+                       .addTag("dynamic")
+                       .configuredProperties(configuredProperties)
+                       .connectorId("sql")
+                       .description("Connection to Sampledb")
+                       .icon("fa-database")
+                       .name("PostgresDB")
+                       .connector(connector)
+                       .build();
+    }
+
+    @Test
+    public void shouldHaveEquivalentObjects() {
+        ConnectorAction action1 = connectorAction("Invoke SQL");
+        Connector connector1 = connector(action1);
+        Connection conn1 = connection(connector1);
+        Integration integration1 = new Integration.Builder()
+            .name("myIntegration")
+            .addConnection(conn1)
+            .build();
+
+        ConnectorAction action2 = connectorAction("Invoke SQL");
+        Connector connector2 = connector(action2);
+        Connection conn2 = connection(connector2);
+        Integration integration2 = new Integration.Builder()
+            .name("myIntegration")
+            .addConnection(conn2)
+            .build();
+
+        Equivalencer equiv = new Equivalencer();
+        boolean equivalent = equiv.equivalent(null, integration1, integration2);
+        assertTrue(equivalent);
+        assertTrue(equiv.message().isEmpty());
+    }
+
+    @Test
+    public void shouldHaveNonEquivalentObjects() {
+        ConnectorAction action1 = connectorAction("Invoke SQL");
+        Connector connector1 = connector(action1);
+        Connection conn1 = connection(connector1);
+        Integration integration1 = new Integration.Builder()
+            .name("myIntegration")
+            .addConnection(conn1)
+            .build();
+
+        ConnectorAction action2 = connectorAction("Invoke PL-SQL");
+        Connector connector2 = connector(action2);
+        Connection conn2 = connection(connector2);
+        Integration integration2 = new Integration.Builder()
+            .name("myIntegration")
+            .addConnection(conn2)
+            .build();
+
+        Equivalencer equiv = new Equivalencer();
+        boolean equivalent = equiv.equivalent(null, integration1, integration2);
+        assertFalse(equivalent);
+        assertEquals(
+                     integration1.getName() + COLON + "Integration" +
+                     SPACE + CLOSE_ANGLE_BRACKET + SPACE +
+                     conn1.getName() + COLON + "Connection" +
+                     SPACE + CLOSE_ANGLE_BRACKET + SPACE +
+                     connector1.getName() + COLON + "Connector" +
+                     SPACE + CLOSE_ANGLE_BRACKET + SPACE +
+                     action1.getName() + COLON + "ConnectorAction" +
+                     DOLLAR_SIGN + "name" + NEW_LINE +
+                     TAB + "=> 'Invoke SQL'" + NEW_LINE +
+                     TAB + "=> 'Invoke PL-SQL'" + NEW_LINE,
+                     equiv.message());
```

Comment:
It is not so easy to understand what message is expected here

Reply:
This is linked to to the previous comment, as I do not know what to expect an example would help understanding it.

Reply:
ok I can change it to a string literal.